### PR TITLE
PLUGIN-219: Pipeline with HttpSink plugin fails to deploy if URL field is specified as macro.

### PR DESCRIPTION
**Pull request by Roni98**
Pipeline with HttpSink plugin fails to deploy if URL field is specified as macro.
JIRA ticket: https://cdap.atlassian.net/browse/PLUGIN-219

### CODE REVIEW

File: src/test/java/io/cdap/plugin/http/sink/batch/HTTPSinkConfigTest.java [hunk 1]
```diff
@@ -67,6 +75,32 @@ public void testInvalidUrl() {
...
+    PluginProperties rawProperties = PluginProperties.builder()
+      .addAll(properties)
+      .build()
+      .setMacros(macros);
+
+    FieldSetter fs = new FieldSetter(config, HTTPSinkConfig.class.getSuperclass().getSuperclass()
+      .getDeclaredField("properties"));
+    fs.set(rawProperties);
+
+
```
Comment (flakrimjusufi) [inline src/test/java/io/cdap/plugin/http/sink/batch/HTTPSinkConfigTest.java:98]: nit: Move one line up.

File: src/test/java/io/cdap/plugin/http/sink/batch/HTTPSinkConfigTest.java [hunk 2]
```diff
@@ -67,6 +75,32 @@ public void testInvalidUrl() {
     assertPropertyValidationFailed(failureCollector, HTTPSinkConfig.URL);
   }
 
+  @Test
+  public void testMacroUrl() throws Exception {
```
Comment (flakrimjusufi) [inline src/test/java/io/cdap/plugin/http/sink/batch/HTTPSinkConfigTest.java]: This is not the approach on how we test with Macro. 

Please take this as a reference and change your test accordingly:

https://github.com/cdapio/hydrator-plugins/blob/29a16d6e9f70447144930be47b85a37c99c1cc87/core-plugins/src/test/java/io/cdap/plugin/batch/sink/TableSinkTest.java#L255

**Review by flakrimjusufi — CHANGES_REQUESTED** [review]

File: src/test/java/io/cdap/plugin/http/sink/batch/HTTPSinkConfigTest.java (hunk 2 above)
Comment (Roni98) [reply src/test/java/io/cdap/plugin/http/sink/batch/HTTPSinkConfigTest.java]: PR updated.

**Review by flakrimjusufi — COMMENTED** [review]
It looks good to me. 

Now you can go into verification process.

File: src/test/java/io/cdap/plugin/http/sink/batch/HTTPSinkTest.java [hunk 3]
```diff
@@ -174,4 +177,69 @@ private int getFeeds() throws IOException {
     urlConn.disconnect();
     return responseCode;
   }
+
+  @Test
+  public void testHTTPSinkMacroUrl() throws Exception {
+    String inputDatasetName = "input-http-sink-with-macro";
+    ETLStage source = new ETLStage("source", MockSource.getPlugin(inputDatasetName));
+    Map<String, String> properties = new ImmutableMap.Builder<String, String>()
+      .put("url", "${url}")
```
Comment (CuriousVini) [inline src/test/java/io/cdap/plugin/http/sink/batch/HTTPSinkTest.java:186]: please add a negative test when url is null or empty

Comment (Roni98) [reply src/test/java/io/cdap/plugin/http/sink/batch/HTTPSinkTest.java:186]: PR updated.

File: src/test/java/io/cdap/plugin/http/sink/batch/HTTPSinkTest.java [hunk 4]
```diff
@@ -174,4 +176,109 @@ private int getFeeds() throws IOException {
...
+      StructuredRecord.builder(inputSchema).set("id", "1").build()
+    );
+    MockSource.writeInput(inputManager, input);
+    // run the pipeline
+    runETLOnce(appManager, runtimeProperties);
+    verifyandcleanupTest();
+  }
+
+  @Test
+  public void testHTTPSinkWithEmptyUrl() {
```
Comment (CuriousVini) [inline src/test/java/io/cdap/plugin/http/sink/batch/HTTPSinkTest.java]: If there is exception that is thrown we should rely on junit for that. We should not silently ignore the exception
`@Test(expected=<SomeException>.class)`

Comment (Roni98) [reply src/test/java/io/cdap/plugin/http/sink/batch/HTTPSinkTest.java]: PR updated accordingly

**Review by CuriousVini — APPROVED** [review]
please squash the commits
